packages/openapi-fetch/src/index.d.ts

```diff
@@ -24,7 +24,7 @@ export interface ClientOptions extends Omit<RequestInit, "headers"> {
 }
 
 export type HeadersOptions =
-  | HeadersInit
+  | Required<RequestInit>["headers"]
```

Comment:
This change pulls a reference to `HeadersInit` from `RequestInit` rather than relying on the global declaration, which only exists in the DOM lib. In projects with the DOM lib, this reference will resolve to `HeadersInit` declared in `lib.dom.d.ts`. In projects with Node types and no DOM lib, this reference will resolve to `HeadersInit` from `undici-types/fetch.d.ts` as referenced from `@types/node/globals.d.ts`.